main.rb

```diff
@@ -0,0 +1,141 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = to_a
+    i = 0
+    while i < array.length
+      yield(array[i])
+      i += 1
+    end
+    array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    array = to_a
+    i = 0
+    while i < array.length
+      yield(array[i], i)
+      i += 1
+    end
+    array
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each { |elem| new_array.push(elem) if yield(elem) == true }
+    new_array
+  end
+
+  def my_all?
+    return to_enum(:my_all?) unless block_given?
+
+    my_each { |elem| return false if yield(elem) == false }
+    true
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    my_each { |elem| return true if yield(elem) == true }
+    false
+  end
```

Comment:
Enumerable `my_any?` when no block or argument is given returns true if at least one of the collection is not false or nil
      Failure/Error: `expect(true_array.my_any?).to be true_array.any?`

        expected true
             got #<Enumerator:2120> => #<Enumerator: [nil, false, true, []]:my_any?>
##
Enumerable `my_any?` when no block or argument is given returns false if at least one of the collection is not true
      Failure/Error: `expect(false_array.my_any?).to be false_array.any?`

        expected false
             got #<Enumerator:2140> => #<Enumerator: [nil, false, nil, false]:my_any?>
##
Enumerable `my_any?` when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
      Failure/Error: `expect(array.my_any?(Numeric)).to be array.any?(Numeric)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_any?` when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
      Failure/Error: `expect(words.my_any?(Integer)).to be words.any?(Integer)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_any?` when a Regex is passed as an argument returns true if any of the collection matches the Regex      Failure/Error: `expect(words.my_any?(/d/)).to be words.any?(/d/)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_any?` when a Regex is passed as an argument returns false if none of the collection matches the Regex
      Failure/Error: `expect(words.my_any?(/z/)).to be words.any?(/z/)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_any?` when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
      Failure/Error: `expect(words.my_any?('cat')).to be words.any?('cat')`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_any?` when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
      Failure/Error: `expect(words.my_any?('cat')).to be words.any?('cat')`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)